src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionsUpdateActionUtils.java

```diff
@@ -175,12 +174,15 @@
                               matchingNewAttributeDefinitionDraft.getAttributeType())) {
                             return buildActions(oldAttributeDefinition, attributeDefinitionDraft);
                           } else {
-                            // since there is no way to change an attribute type on CTP,
-                            // we remove the attribute definition and add a new one with a new
-                            // attribute type
-                            return Arrays.asList(
-                                RemoveAttributeDefinition.of(oldAttributeDefinitionName),
-                                AddAttributeDefinition.of(attributeDefinitionDraft));
+                            throw new UnsupportedOperationException(
+                                format(
+                                    "Due to eventual consistency of 'removeAttributeDefinition' action, "
+                                        + "changing the attribute definition type (name='%s') is not "
```

Comment:
```suggestion
                                        + "changing the '%s' attribute definition type is not "
```

Reply:
I think the previous one is more descriptive as it points to name of the attribute definition.

Reply:
In the previous version it is not clear if you the name of the product type or the attribute itself

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/787/commits/54368c66f40d3093ed9979dbd52757b3966547d0